server/api/users.js

```diff
@@ -1,60 +1,94 @@
-const router = require('express').Router();
-const { User } = require('../db/models');
-module.exports = router;
+const router = require('express').Router()
+const {User} = require('../db/models')
+const {adminMiddleware, isUserOrAdminMiddleware} = require('./middleware')
 
-router.get('/', async (req, res, next) => {
+module.exports = router
+
+router.get('/', adminMiddleware, async (req, res, next) => {
   try {
     const users = await User.findAll({
       // explicitly select only the id and email fields - even though
       // users' passwords are encrypted, it won't help if we just
       // send everything to anyone who asks!
-      attributes: ['id', 'email'],
-    });
-    res.json(users);
+      attributes: ['id', 'email']
+    })
+    res.json(users)
   } catch (err) {
-    next(err);
+    next(err)
   }
-});
+})
 
-router.get('/:userId', async (req, res, next) => {
+router.get('/:userId', isUserOrAdminMiddleware, async (req, res, next) => {
```

Comment:
another way to get around having this weird combo middleware is to have the api call different routes. maybe there should be an /admin api that does all the admin actions, and edits to a user by themselves are routed through /user as expected. 

That feels cleaner to me and in the future, if we want to do some logging/change history, it's clearer what happened by who, and how.

Reply:
routes have been separated, see updated changes